velero-csi-plugin/csi_snapshotter.go

```diff
@@ -117,10 +140,40 @@ func (p *CSISnapshotter) Execute(item runtime.Unstructured, backup *v1.Backup) (
 	if annotations == nil {
 		annotations = make(map[string]string)
 	}
-	annotations["velero.io/volume-snapshot-name"] = upd.Name
+	annotations[veleroSnapshotNameAnnotation] = upd.Name
+	annotations[veleroBackupNameAnnotation] = backup.Name
 	accessor.SetAnnotations(annotations)
 
-	return item, nil, nil
+	//TODO: add retry logic here. CSI driver may take time to reconcile the snapshotcontents object
+	snapshotContents, err := getVolumeSnapshotContentForVolumeSnapshot(snapshotClient.SnapshotV1beta1(), upd.Name)
+
+	// TODO: Create GroupResource obejects for volumesnapshot objects
+	additionalItems := []velero.ResourceIdentifier{
+		{
+			GroupResource: schema.GroupResource{
```

Comment:
We could probably define these in core Velero's `pkg/kuberesource` package.

Reply:
Yes, that was the idea. But, I put it here for now.

Reply:
Opened this PR https://github.com/vmware-tanzu/velero/pull/2288